src/share/poudriere/common.sh

```diff
@@ -5280,7 +5283,7 @@ delete_old_pkg() {
 	# Check if the compiled options match the current options from make.conf and /var/db/ports
 	if [ "${CHECK_CHANGED_OPTIONS}" != "no" ]; then
 		if have_ports_feature SELECTED_OPTIONS; then
-			shash_remove pkgname-options "${new_pkgname}" \
+			shash_get pkgname-options "${new_pkgname}" \
```

Comment:
Is this intentional? The `shash_remove` fetches and removes the cache item. The `shash_get` only fetches. This cache is not persistent across builds. It's just for IPC.

Reply:
Sorry that it's not documented.

Reply:
Yes, because we now have 2 copies of every package in the repo directory, the second one was not finding the cached options, and it was triggering rebuilds for 'changed options'. This change means that the 2nd copy of the pkg will still find the options in the cache.

Reply:
Well doubling the startup time isn't wanted and this will be hard to maintain. Perhaps symlinks should be trimmed out (by changing the glob *.PKG_EXT to a `$(cd dir && find . ...)` kind of thing) or % files should be skipped in `delete_old_pkg`.

Do we even need the symlinks?

Reply:
I wanted to not have the symlink, but that was causing a lot of other fallout.

Since now the filename is unpredictable, things like `clean_build_queue()` can only look for the symlink, since they cannot guess the hashed name of the file.

Reply:
The need for this has been overcome by moving the hashed files into the .hashed directory, to avoid duplicate detection